pandas/io/tests/test_pytables.py

```diff
@@ -930,6 +931,47 @@ def test_encoding(self):
             result = store.select('df',Term('columns=A',encoding='ascii'))
             tm.assert_frame_equal(result,expected)
 
+    def test_latin_encoding(self):
+
+        if compat.PY2:
+            self.assertRaisesRegexp(TypeError, '\[unicode\] is not implemented as a table column')
+            return
+
+        values = [[b'E\xc9, 17', b'', b'a', b'b', b'c'],
+                  [b'E\xc9, 17', b'a', b'b', b'c'],
+                  [b'EE, 17', b'', b'a', b'b', b'c'],
+                  [b'E\xc9, 17', b'\xf8\xfc', b'a', b'b', b'c'],
+                  [b'', b'a', b'b', b'c'],
+                  [b'\xf8\xfc', b'a', b'b', b'c'],
+                  [b'A\xf8\xfc', b'', b'a', b'b', b'c'],
+                  [np.nan, b'', b'b', b'c'],
+                  [b'A\xf8\xfc', np.nan, b'', b'b', b'c']]
+
+        def _try_decode(x, encoding='latin-1'):
+            try:
+                return x.decode(encoding)
+            except AttributeError:
+                return x
+        # not sure how to remove latin-1 from code in python 2 and 3
+        values = [[_try_decode(x) for x in y] for y in values]
+
+        examples = []
+        for dtype in ['category', object]:
+            for val in values:
+                examples.append(pandas.Series(val, dtype=dtype))
+
+        def roundtrip(s, key='data', encoding='latin-1', nan_rep=''):
+            with ensure_clean_path(self.path) as store:
+                s.to_hdf(store, key, format='table', encoding=encoding,
+                        nan_rep=nan_rep)
+                retr = read_hdf(store, key)
+                s_nan = s.replace('', np.nan)
+                assert_series_equal(s_nan, retr)
+
+        for s in examples:
```

Comment:
Seems to hit problems in the type inferring.

roundtrip(s, nan_rep=b'\xf8\xfc') fails with the following error:

======================================================================
ERROR: test_latin_encoding (pandas.io.tests.test_pytables.TestHDFStore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/tests/test_pytables.py", line 974, in test_latin_encoding
    roundtrip(s, nan_rep=b'\xf8\xfc')
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/tests/test_pytables.py", line 965, in roundtrip
    nan_rep=nan_rep)
  File "/Users/davidcottrell/dev/pandas.git/pandas/core/generic.py", line 931, in to_hdf
    return pytables.to_hdf(path_or_buf, key, self, **kwargs)
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/pytables.py", line 269, in to_hdf
    f(store)
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/pytables.py", line 264, in <lambda>
    f = lambda store: store.put(key, value, **kwargs)
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/pytables.py", line 826, in put
    self._write_to_group(key, value, append=append, **kwargs)
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/pytables.py", line 1275, in _write_to_group
    s.write(obj=value, append=append, complib=complib, **kwargs)
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/pytables.py", line 4101, in write
    obj=obj, data_columns=obj.columns, **kwargs)
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/pytables.py", line 3799, in write
    **kwargs)
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/pytables.py", line 3476, in create_axes
    raise e
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/pytables.py", line 3471, in create_axes
    **kwargs)
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/pytables.py", line 1873, in set_atom
    encoding)
  File "/Users/davidcottrell/dev/pandas.git/pandas/io/pytables.py", line 1903, in set_atom_string
    % (item, inferred_type)
TypeError: Cannot serialize the column [values] because
its data contents are [mixed] object dtype